File: 2021-09-01/JannaKim.py
v = "AEIOU"


def combine(* args):
    res = ''
    for idx in args:
        res += v[idx] 
    return res
        
def solution(word):
    cnt = 0
        
    for i in range(5):
        cnt += 1
        res = combine(i)
        
        if res == word:
            return cnt
        
        for j in range(5):
            cnt += 1
            res = combine(i, j)
            
            if res == word:
                return cnt
                
            for k in range(5):
                cnt += 1
                res = combine(i, j, k)
                
                if res == word:
                    return cnt
                    
                for l in range(5):
                    cnt += 1
                    res = combine(i, j, k, l)
                    
                    if res == word:
                        return cnt
                        
                    for m in range(5):
                        cnt += 1
                        res = combine(i, j, k, l, m)
                        
                        if res == word:
                            return cnt
                        
    return -1

File: 2021-09-01/JannaKim.cpp
#include <string>
#include <iostream>
using namespace std;

string v = "AEIOU";
string w;
string occ = "";
int ans = -1;

bool dfs(int depth); // occurence

int solution(string word) {
    w = word; //
    dfs(0);
    return ans;
}

bool dfs(int depth){
    ++ans;
    if(occ == w) return true; // compare cur with given
    if(depth == 5) return false;

    for(int i = 0; i < 5; ++i){
        occ.push_back(v[i]); // backtrack
        if(dfs(depth + 1)) return true;
        occ.pop_back();
    }
    return false;
}